"""fix for wrong type of foreign keys in associations

Revision ID: 2c74fcb63719
Revises: 
Create Date: 2023-07-13 23:46:24.389618

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '2c74fcb63719'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('building_cards_in_deck', schema=None) as batch_op:
        batch_op.alter_column('room',
               existing_type=sa.VARCHAR(length=32),
               type_=sa.Integer(),
               existing_nullable=False)

    with op.batch_alter_table('magic_cards_in_deck', schema=None) as batch_op:
        batch_op.alter_column('room',
               existing_type=sa.VARCHAR(length=32),
               type_=sa.Integer(),
               existing_nullable=False)

    with op.batch_alter_table('soldiers_cards_in_deck', schema=None) as batch_op:
        batch_op.alter_column('room',
               existing_type=sa.VARCHAR(length=32),
               type_=sa.Integer(),
               existing_nullable=False)

    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('soldiers_cards_in_deck', schema=None) as batch_op:
        batch_op.alter_column('room',
               existing_type=sa.Integer(),
               type_=sa.VARCHAR(length=32),
               existing_nullable=False)

    with op.batch_alter_table('magic_cards_in_deck', schema=None) as batch_op:
        batch_op.alter_column('room',
               existing_type=sa.Integer(),
               type_=sa.VARCHAR(length=32),
               existing_nullable=False)

    with op.batch_alter_table('building_cards_in_deck', schema=None) as batch_op:
        batch_op.alter_column('room',
               existing_type=sa.Integer(),
               type_=sa.VARCHAR(length=32),
               existing_nullable=False)

    # ### end Alembic commands ###
